client/gutenberg/extensions/presets/jetpack/utils/register-jetpack-block.js

```diff
@@ -2,28 +2,34 @@
 /**
  * External dependencies
  */
-import { get } from 'lodash';
 import { registerBlockType } from '@wordpress/blocks';
 
 /**
  * Internal dependencies
  */
-import getJetpackData from './get-jetpack-data';
+import isJetpackExtensionAvailable from './is-jetpack-extension-available';
 
 /**
  * Registers a gutenberg block if the availability requirements are met.
  *
  * @param {string} name The block's name.
  * @param {object} settings The block's settings.
+ * @param {object} childBlocks The block's child blocks.
  * @returns {object|false} Either false if the block is not available, or the results of `registerBlockType`
  */
-export default function registerJetpackBlock( name, settings ) {
-	const data = getJetpackData();
-	const available = get( data, [ 'available_blocks', name, 'available' ], false );
-	if ( data && ! available ) {
+export default function registerJetpackBlock( name, settings, childBlocks = [] ) {
+	if ( ! isJetpackExtensionAvailable( name ) ) {
 		// TODO: check 'unavailable_reason' and respond accordingly
 		return false;
 	}
 
-	return registerBlockType( `jetpack/${ name }`, settings );
+	const result = registerBlockType( `jetpack/${ name }`, settings );
+
+	// Register child blocks. Using `registerBlockType()` directly avoids availability checks -- if
+	// their parent is available, we register them all, without checking for their individual availability.
+	childBlocks.forEach( childBlock =>
+		registerBlockType( `jetpack/${ childBlock.name }`, childBlock.settings )
```

Comment:
What is the process of de registering block on the calypso side? Is that something we should be worried about now?